File: app/src/main/java/de/tum/in/tumcampusapp/models/tumcabe/RoomFinderRoom.java
package de.tum.in.tumcampusapp.models.tumcabe;

import java.io.Serializable;

import de.tum.in.tumcampusapp.adapters.SimpleStickyListHeadersAdapter.SimpleStickyListItem;

/**
 * This class is used as a model for rooms in Roomfinder retrofit request.
 */
public class RoomFinderRoom implements SimpleStickyListItem, Serializable {
    private static final long serialVersionUID = 6631656320611471476L;

    String campus;
    String address;
    String info;
    String arch_id;
    String room_id;
    String name;    // Campus name

    public RoomFinderRoom(String campus, String address, String info, String arch_id, String room_id, String name) {
        this.campus = campus;
        this.address = address;
        this.info = info;
        this.arch_id = arch_id;
        this.room_id = room_id;
        this.name = name;
    }

    public String getCampus() {
        return campus;
    }

    public String getAddress() {
        return address;
    }

    public String getInfo() {
        return info;
    }

    public String getArch_id() {
        return arch_id;
    }

    public String getRoom_id() {
        return room_id;
    }

    public String getName() {
        if (name != null && name.equals("null")) {
            return "";
        }
        return name;
    }

    @Override
    public String getHeadName() {
        return getName();
    }

    @Override
    public String getHeaderId() {
        return getHeadName();
    }

    @Override
    public String toString() {
        return getCampus() + ";" + getAddress() + ";"
               + getInfo() + ";" + getArch_id() + ";"
               + getRoom_id() + ";" + getName();
    }
}

File: app/src/main/java/de/tum/in/tumcampusapp/models/tumcabe/RoomFinderRoom.kt
package de.tum.`in`.tumcampusapp.models.tumcabe

import de.tum.`in`.tumcampusapp.adapters.SimpleStickyListHeadersAdapter.SimpleStickyListItem
import de.tum.`in`.tumcampusapp.models.dbEntities.Recent
import java.io.Serializable

/**
 * This class is used as a model for rooms in Roomfinder retrofit request.
 * @param name This is the campus name
 */
data class RoomFinderRoom(var campus: String = "",
                          var address: String = "",
                          var info: String = "",
                          var arch_id: String = "",
                          var room_id: String = "",
                          private val name: String = "") : SimpleStickyListItem, Serializable {

    fun getName(): String =
            if (name == "null")
                ""
            else
                name


    override fun getHeadName() = getName()

    override fun getHeaderId() = headName

    companion object {
        private const val serialVersionUID = 6631656320611471476L

        fun fromRecent(r: Recent): RoomFinderRoom {
            val values = r.name.split(";".toRegex()).dropLastWhile { it.isEmpty() }.toTypedArray()
            if (values.size != 6) {
                throw IllegalArgumentException()
            }
            return RoomFinderRoom(values[0], values[1], values[2], values[3], values[4], values[5])
        }
    }
}
